lib/matrix_check_sum.rb

```diff
@@ -3,8 +3,30 @@
 # of numbers in row i is the same as the sum of numbers in column i for i = 0 to row.length-1
 # If this is the case, return true. Otherwise, return false.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2), where n is the number of rows (since # of columns and rows are assumed to be the same)
+# Space complexity: O(n), where n is the number of rows
```

Comment:
Your solution is doing this as O(n) space complexity, but you **could** do this with space complexity O(1).  Look at how you have separate loops for adding up the row and column sums, do you have to do it separately?